app/views/questionnaire.py

```diff
@@ -150,22 +200,110 @@ def post_block(routing_path, schema, metadata, collection_metadata, answer_store
         answer_store_updater.save_answers(form)
+        if form.data[block['add_answer']['id']] == block['add_answer']['value']:
+            add_url = url_for('questionnaire.get_add_list_item', list_name=rendered_block['populates_list'], add_block_id=rendered_block['add_block']['id'])
+            return add_url
+        return
 
-        next_location = path_finder.get_next_location(current_location=current_location)
+    if block['type'] == 'ListRemoveQuestion':
+        if form.data[parent_block['remove_answer']['id']] == parent_block['remove_answer']['value']:
+            list_name = parent_block['populates_list']
+            answer_store_updater.remove_all_answers_with_list_item_id(list_name, list_item_id)
+        else:
+            return url_for('questionnaire.get_block', block_id=parent_block['id'])
 
-        if _is_end_of_questionnaire(block, next_location):
-            return submit_answers(routing_path, schema)
+    if block['type'] == 'ListAddQuestion':
+        answer_store_updater.save_new_list_item_answers(form, parent_block['populates_list'])
+    elif block['type'] == 'ListEditQuestion':
+        answer_store_updater.save_answers(form, save_completed_blocks=False)
 
-        return redirect(next_location.url())
+    # Clear the answer from the confirmation question on the list collector question
+    transformed_parent = transform_variants(parent_block, schema, metadata, answer_store)
+    answer_ids_to_remove = [answer['id'] for answer in transformed_parent['question']['answers']]
+    answer_store_updater.remove_answer_ids(answer_ids_to_remove)
 
-    context = build_view_context(block['type'], metadata, schema, answer_store, rendered_block, current_location, form)
+    list_collector_url = url_for('questionnaire.get_block', block_id=parent_block['id'])
 
-    return _render_page(block['type'], context, current_location, schema)
+    return list_collector_url
+
+
+@questionnaire_blueprint.route('<block_id>', methods=['POST'])
+@login_required
+@with_questionnaire_store
+@with_schema
+@full_routing_path_required
+# pylint: disable=too-many-locals, too-many-return-statements
+def post_block(routing_path, schema, questionnaire_store, block_id):
+
+    return post_block_handler(
+        routing_path,
+        schema,
+        questionnaire_store,
+        block_id,
+    )
+
+
+@questionnaire_blueprint.route('<list_name>/<block_id>', methods=['POST'])
+@login_required
+@with_questionnaire_store
+@with_schema
+def post_add_list_item(schema, questionnaire_store, list_name, block_id):
+    return post_block_handler(
+        None,
+        schema,
+        questionnaire_store,
+        block_id,
+        list_name,
+        None,
+    )
+
+
+@questionnaire_blueprint.route('<list_name>/<list_item_id>/<block_id>', methods=['POST'])
+@login_required
+@with_questionnaire_store
+@with_schema
+def post_list_item_block(schema, questionnaire_store, list_name, list_item_id, block_id):
+    return post_block_handler(
+        None,
```

Comment:
I think the routing path is required for these methods as we'll need to validate that they can get to the list collector for the add/change/remove block they're posting.